codegen/core/src/test/scala/com/lightbend/akkasls/codegen/ModelBuilderSuite.scala

```diff
@@ -345,77 +804,289 @@ abstract class ModelBuilderSuite(config: ModelBuilderSuite.Config) extends munit
+      // View components don't have a proto package.
+      // So we need to build one for them since they are declared in a Service annotation,
+      // we stay with this package.
+      val derivedShoppingCartPackage = shoppingCartPackage.asJavaMultiFiles
 
       val transformedUpdates =
         List(
           command(
             "ProcessAdded",
-            fullyQualifiedName("ItemAdded", domainProto),
-            fullyQualifiedName("CartViewState", shoppingCartProto)),
+            fullyQualifiedName("ItemAdded", domainPackage),
+            fullyQualifiedName("CartViewState", shoppingCartPackage)),
           command(
             "ProcessRemoved",
-            fullyQualifiedName("ItemRemoved", domainProto),
-            fullyQualifiedName("CartViewState", shoppingCartProto)),
+            fullyQualifiedName("ItemRemoved", domainPackage),
+            fullyQualifiedName("CartViewState", shoppingCartPackage)),
           command(
             "ProcessCheckedOut",
-            fullyQualifiedName("CheckedOut", domainProto),
-            fullyQualifiedName("CartViewState", shoppingCartProto)))
+            fullyQualifiedName("CheckedOut", domainPackage),
+            fullyQualifiedName("CartViewState", shoppingCartPackage)))
 
       val queries = List(
         command(
           "GetCheckedOutCarts",
-          fullyQualifiedName("GetCheckedOutCartsRequest", shoppingCartProto),
-          fullyQualifiedName("CartViewState", shoppingCartProto),
+          fullyQualifiedName("GetCheckedOutCartsRequest", shoppingCartPackage),
+          fullyQualifiedName("CartViewState", shoppingCartPackage),
           streamedOutput = true))
       assertEquals(
         model.services,
         Map(
-          "com.example.shoppingcart.view.ShoppingCartViewService" ->
+          "com.example.shoppingcart.view.ShoppingCartView" ->
           ModelBuilder.ViewService(
+            // this is the name as defined in the proto file
             FullyQualifiedName(
-              "ShoppingCartViewService",
-              "ShoppingCartViewService",
-              shoppingCartProto,
-              TestData.guessDescriptor("ShoppingCartViewService", shoppingCartProto)),
+              "ShoppingCartView",
+              "ShoppingCartView",
+              derivedShoppingCartPackage,
+              TestData.guessDescriptor("ShoppingCartViewService", shoppingCartPackage)),
             transformedUpdates ++ queries,
             "ShoppingCartViewService",
             transformedUpdates,
             transformedUpdates,
-            queries)))
+            queries,
+            transformName = false)))
     }.get
   }
+}
 
-  test(s"deriving java package from proto options (${config.label})") {
-    val protoFileName = "name.proto"
-    val name = "Name"
-    val pkg = "com.example"
+class ModelBuilderWithServiceAnnotationSuite extends ModelBuilderSuite(ModelBuilderSuite.ServiceAnnotationConfig) {
```

Comment:
I wish I could have kept the three tests below under `ModelBuilderSuite` and not duplicate them, but the are slightly different from the new codegen annotation one because of the choice of descriptors.